packages/enzyme/src/ShallowWrapper.js

```diff
@@ -387,7 +387,7 @@ class ShallowWrapper {
         // so we replace shouldComponentUpdate to know the result and restore it later.
         let originalShouldComponentUpdate;
         if (
-          this[OPTIONS].lifecycleExperimental &&
```

Comment:
hm, i wonder if this is causing a number of other bugs related to lifecycle methods?

Reply:
I can't find issues causing by this in enzyme repository, but there might be in outside.